src/java/test/org/apache/zookeeper/server/quorum/QuorumPeerMainTest.java

```diff
@@ -615,14 +631,18 @@ public void testQuorumDefaults() throws Exception {
         ClientBase.setupTestEnv();
 
         // setup the logger to capture all logs
+        LoggerContext loggerContext =  (LoggerContext) LogManager.getContext(false);
+        Configuration logConfig = loggerContext.getConfiguration();
+        
         Layout layout =
-                Logger.getRootLogger().getAppender("CONSOLE").getLayout();
+        		logConfig.getRootLogger().getAppenders().get("STDOUT").getLayout();
+        String appenderName = "quorumAppender";
         ByteArrayOutputStream os = new ByteArrayOutputStream();
-        WriterAppender appender = new WriterAppender(layout, os);
-        appender.setImmediateFlush(true);
-        appender.setThreshold(Level.INFO);
-        Logger zlogger = Logger.getLogger("org.apache.zookeeper");
-        zlogger.addAppender(appender);
+        WriterAppender appender = WriterAppender.createAppender((StringLayout)layout, null, new OutputStreamWriter(os), appenderName, false, true);
```

Comment:
It's probably a good idea to use the builder to create the appender (see above).